### ISSUE TITLE
[BUG] Iterating over result of ShareServiceClient.listShares results in infinite loop

### ISSUE THREAD

**Issue by willemv** [Azure/azure-sdk-for-java#18880]
**Describe the bug**
Calling `ShareServiceClient.listShares` and iterating over the results endlessly loops over all shares.

**To Reproduce**
* Set up an Azure Storage account
* Add a limited amount of shares (e.g. two)
* create a `ShareServiceClient`
* call `listShares` on the client, and iterate over the result

***Code Snippet***
Using `stream()`
``` 
ShareServiceClient azureClient = new ShareServiceClientBuilder()
        .connectionString(connectionString)
        .buildClient();
azureClient.listShares()
           .stream()
           .forEach(shareItem -> System.out.println("Share: " + shareItem.getName()));
```

Using the for-each construct:
```
for (ShareItem shareItem : azureClient.listShares()) {
    System.out.println("Share: " + shareItem.getName());
}
```

Using the paged functionality:
```
for (PagedResponse<ShareItem> page : azureClient.listShares().iterableByPage(100)) {
    for (ShareItem shareItem : page.getValue()) {
        System.out.println("Share:" + shareItem.getName());
    }
}
```

**Expected behavior**
The name of each available share is printed once

Actual result: this keeps printing the same share names over and over, without ever stopping.

**Setup (please complete the following information):**
 - OS: Linux, macOS, Windows
 - IDE : IntelliJ, Eclipse, gradle, maven, ...
 - Version of the Library used: 
    - `azure-storage-file-share` : `12.8.0`
    - `azure-storage-common` : `12.10.0`
    - `azure-core`: `1.12.0`

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by rickle-msft** [comment]
@willemv Thank you for reporting this issue. I suspect this is the same root cause as the other issue you filed, so let's continue the conversation over there to keep everything centralized.

**Comment by gapra-msft** [comment]
This was recently fixed by the release of azure-core. We should transitively pick up the fix once we release our next version of storage

**Comment by gapra-msft** [comment]
Hi @willemv, we recently released azure-storage-file-share 12.9.0-beta.2 that should fix this issue. Please try it out and let us know if you are still hitting the issue. 

Feel free to close this once you've confirmed the latest release fixes your issue.

**Comment by gapra-msft** [comment]
Hi @willemv, are you still hitting this issue with the latest release?

**Comment by willemv** [comment]
The issue is fixed in 12.9.0-beta.2 and later, thanks
